lib/taurus/qt/qtgui/panel/taurusvalue.py

```diff
@@ -1165,13 +1165,14 @@ def createConfig(self, allowUnpickable=False):
                     or allowUnpickable):
                 #configdict[key] = classID
                 configdict[key] = {'classid': classID}
-                widget = getattr(self, key[0].lower() + key[1:])()
-                if isinstance(widget, BaseConfigurableClass):
-                    configdict[key]['delegate'] = widget.createConfig()
             else:
-                self.info('createConfig: %s not saved because it is not Pickable (%s)' % (
+                self.debug('createConfig: classid of %s not saved because it is not Pickable (%s)' % (
                     key, str(classID)))
-
+            widget = getattr(self, key[0].lower() + key[1:])()
+            if isinstance(widget, BaseConfigurableClass):
+                widget_configdict = configdict.get(key, {})
+                widget_configdict['delegate'] = widget.createConfig()
+                configdict[key] = widget_configdict
```

Comment:
I am not sure about this change: by moving the configdict creation outside the "if", potentially unpickable configs would be created **even when `allowUnpickable == False`**, which I do not think is desirable. 

If a given widget (e.g. `PoolMotorTV`) can handle unpickable configs, maybe the whole issue is solved by simply defining `allowUnpickable=True` for that widget...

Reply:
I correct myself: no unpickable configs would be created, but still there would be an issue when calling applyConfig with this config in some situations if classid is not stored.